duckduckgo/macos-browser
# Punycode support

Task/Issue URL: https://app.asana.com/0/72649045549333/1200034801624474
Tech Design URL:
CC: @brindy 

**Description**:
This PR adds Punycode support

**Steps to test this PR**:
0. use test urls "82.мвд.рф", "жкхнсо.рф", "💩.la", "💩.la:8080"
1. suggestions should show readable addresses
2. "– Visit address" suffix should show readable address
3. punycode encoded addresses should be navigatable
4. passive address bar should display readable address
5. address should be bookmarkable
6. when navigating to a Punycode-encoded url and no page title yet, Tab title should show readable address
7. on Browsing Session Restoration Tab titles should show readable addresses
8. Punycode-encoded url should be editable in readable form
9. Fireproofing should work and display readable names
10. Tab Preview Tooltips should display readable address


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
